pkg/util/common.go

```diff
@@ -171,26 +171,42 @@ func CreateSecretVolume(volumeName string, secretName string, defaultMode int) (
 }
 
 // CreatePod will create the pod
-func CreatePod(name string, serviceAccount string, volumes []*components.Volume, containers []*Container, initContainers []*Container, affinityConfigs []horizonapi.PodAffinityConfig, labels map[string]string) (*components.Pod, error) {
+func CreatePod(name string, podConfig *PodConfig) (*components.Pod, error) {
+	// create pod config
 	pod := components.NewPod(horizonapi.PodConfig{
 		Name: name,
 	})
 
-	if !strings.EqualFold(serviceAccount, "") {
-		pod.Spec.ServiceAccountName = serviceAccount
+	// set service account
+	if !strings.EqualFold(podConfig.ServiceAccount, "") {
```

Comment:
Probably cleaner to just check the length

Reply:
done